### ISSUE TITLE
UITextfield's setDelegate Method expects a UIScrollViewDelegate and not a UITextViewDelegate

### ISSUE THREAD

**Issue by jessemx109** [ReactiveX/RxSwift#1578]
**Short description of the issue**:

 The setDelegate() method on UITextview does not allow you to pass a UITextViewDelegate in, instead it expects a UIScrollviewDelegate. This means you are unable to forward UITextViewDelegate events out of RxCocoa

**Expected outcome**:

  The setDelegate() should take a UITextViewDelegate as the input object

**What actually happens**:

   The setDelegate() expects a UIScrollviewDelegate to be passed in

**Self contained code example that reproduces the issue**:
  This function expects self to be a UIScrollviewDelegate and not a UITextViewDelegate
```swift
  textView.rx.setDelegate(self)

```

**RxSwift/RxCocoa/RxBlocking/RxTest version/commit**

  RxCocoa 4.1.2

**Platform/Environment**

  - [x ] iOS
  - [ ] macOS
  - [ ] tvOS
  - [ ] watchOS
  - [ ] playgrounds

**How easy is to reproduce? (chances of successful reproduce after running the self contained code)**

  - [ x] easy, 100% repro
  - [ ] sometimes, 10%-100%
  - [ ] hard, 2% - 10%
  - [ ] extremely hard, %0 - 2%

**Xcode version**:

```
  Xcode 9.2
```

**Installation method**:
  - [ x] CocoaPods
  - [ ] Carthage
  - [ ] Git submodules

**I have multiple versions of Xcode installed**:
  (so we can know if this is a potential cause of your issue)
  - [ ] yes (which ones)
  - [ x] no

**Level of RxSwift knowledge**:
  (this is so we can understand your level of knowledge
    and formulate the response in an appropriate manner)
  - [ ] just starting
  - [ ] I have a small code base
  - [x ] I have a significant code base

**Comment by kzaher** [comment]
Hi @jessemx109 ,

`UITextViewDelegate` inherits from `UIScrollViewDelegate`.

```swift
public protocol UITextViewDelegate : NSObjectProtocol, UIScrollViewDelegate {

    
    @available(tvOS 2.0, *)
    optional public func textViewShouldBeginEditing(_ textView: UITextView) -> Bool

```

**Comment by jessemx109** [comment]
It may inherit from UIScrollViewDelegate but it still doesnt allow me to override the UITextViewDelegate methods. I need to handle the textShouldChange delegate method
